pkg/listeners/listeners_unix.go

```diff
@@ -31,7 +31,12 @@ func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) ([]net.Listene
 		}
 		ls = append(ls, l)
 	case "unix":
-		l, err := sockets.NewUnixSocket(addr, socketGroup)
+
+		gid, err := strconv.Atoi(socketGroup)
```

Comment:
I think this really needs to be updated such that the function takes a gid as an integer, and then do group lookup in `cmd/dockerd/daemon.go`